src/runtime-rs/crates/hypervisor/src/utils.rs

```diff
@@ -58,15 +58,19 @@ pub fn get_jailer_root(sid: &str) -> String {
     [&sandbox_path, JAILER_ROOT].join("/")
 }
 
-// Clear the O_CLOEXEC which is set by default by Rust standard library
-// as it would obviously prevent passing the descriptor to the hypervisor process.
-pub fn clear_fd_flags(rawfd: RawFd) -> Result<()> {
-    if let Err(err) = fcntl::fcntl(rawfd, fcntl::FcntlArg::F_SETFD(fcntl::FdFlag::empty())) {
-        info!(
-            sl!(),
-            "couldn't clear O_CLOEXEC on device's fd, communication with agent will not work: {:?}",
-            err
-        );
+// Clear the O_CLOEXEC which is set by default by Rust standard library on
+// file descriptors that it opens.  This function is mostly meant to be
+// called on descriptors to be passed to a child (hypervisor) process as
+// O_CLOEXEC would obviously prevent that.
+pub fn clear_cloexec(rawfd: RawFd) -> Result<()> {
+    let cur_flags = fcntl::fcntl(rawfd, fcntl::FcntlArg::F_GETFD)?;
+    let mut new_flags = fcntl::FdFlag::from_bits(cur_flags).ok_or(anyhow!(
+        "couldn't construct FdFlag from flags value {:?}",
+        cur_flags
+    ))?;
+    new_flags.remove(fcntl::FdFlag::FD_CLOEXEC);
+    if let Err(err) = fcntl::fcntl(rawfd, fcntl::FcntlArg::F_SETFD(new_flags)) {
```

Comment:
I agree that this might look a bit pedantic.  Still, not everybody is necessarily aware of how many possible flags there are (I know I wasn't until fairly recently ;-)), so I considered it better to state the intent directly, for the benefit of those readers if for no other reason.